src/Frontend/Views/Portal/PortalHistoryView.tsx

```diff
@@ -19,33 +22,23 @@ export interface RoundHistoryItem {
 }
 
 export const PortalHistoryView: React.FC<{}> = ({}) => {
-  const [rounds, setRounds] = useState<RoundHistoryItem[]>([]);
   const history = useHistory();
   const twitters = useTwitters() as Object;
 
-  useEffect(() => {
-    async function getRoundHistory() {
-      const rounds = await fetch('http://localhost:3000/rounds', {
-        method: 'GET',
-        headers: { 'Content-Type': 'application/json' },
-      });
-      const roundText = await rounds.text();
-      return JSON.parse(roundText).body;
-    }
-    getRoundHistory().then((rounds) => {
-      rounds.forEach((round: any) => {
-        const roundToAdd = {
-          configHash: round.configHash,
-          name: getConfigName(round.configHash),
-          startTime: round.startTime,
-          endTIme: round.endTime,
-          winner: round.winner,
-        };
-        setRounds((prevRounds) => [...prevRounds, roundToAdd]);
-      });
+  const { data: adminData, error } = useSWR(`${process.env.DFDAO_WEBSERVER_URL}/rounds`, fetcher);
+  let rounds = undefined;
+  if (adminData) {
+    rounds = adminData.map((round: any) => {
+      return {
+        configHash: round.configHash,
+        name: getConfigName(round.configHash),
+        startTime: round.startTime,
+        endTIme: round.endTime,
+        winner: round.winner,
+      };
     });
```

Comment:
cleaner if `rounds` is a value from the `useMemo` hook and the logic that maps `adminData` to `rounds` should be inside that hook with `adminData` in the dependency array

Reply:
```tsx
  const rounds = useMemo(() => {
    if (adminData) {
      return adminData.map((round: RoundHistoryItem & { endTime: number }) => {
        return {
          configHash: round.configHash,
          name: getConfigName(round.configHash),
          startTime: round.startTime,
          winner: round.winner,
        };
      });
    } else {
      return [];
    }
  }, [adminData]);
```